windows/src/engine/kmcomapi/processes/keyboard/Keyman.System.Process.KPInstallKeyboardLanguage.pas

```diff
@@ -163,93 +165,93 @@ function TKPInstallKeyboardLanguage.AreMaximumTransientLanguagesInstalled: Boole
 end;
 
 // TODO: change LangID to Winapi.Windows.LANGID
-function TKPInstallKeyboardLanguage.FindInstallationLangID(const BCP47Tag: string; var LangID: Integer; var TemporaryLayoutString: string; Flags: TKPInstallKeyboardLanguageFlags): Boolean;
+function TKPInstallKeyboardLanguage.FindInstallationLangID(var BCP47Tag: string; var LangID: Integer; var TemporaryLayoutString: string; Flags: TKPInstallKeyboardLanguageFlags): Boolean;
 var
   ml: TMitigateWin10_1803.TMitigatedLanguage;
   Win8Lang: TWindows8Language;
-  tag: string;
 begin
   if BCP47Tag = '' then
     ErrorFmt(KMN_E_ProfileInstall_InvalidBCP47Tag, VarArrayOf([BCP47Tag]));
 
-  tag := BCP47Tag;
   TemporaryLayoutString := '';
 
-  if ConvertBCP47TagToLangID(tag, LangID) then
+  if ConvertBCP47TagToLangID(BCP47Tag, LangID) then
   begin
     if TMitigateWin10_1803.IsMitigationRequired(LangID, ml) then
     begin
       LangID := ml.NewLanguage.Code;
+      BCP47Tag := ml.NewLanguage.BCP47;
       WarnFmt(KMN_W_ProfileInstall_Win10_1803_MitigationApplied, VarArrayOf([ml.OriginalLanguage.Name, ml.NewLanguage.Name]));
     end;
-  end
-  else
-  begin
-    if not (ilkInstallTransientLanguage in Flags) then
-    begin
-      // No warning, because this is to be expected
-      Exit(False);
-    end;
 
-    //
-    // Installing a custom language only supported with Win8 and later
-    //
+    if LangID <> 0 then
+      Exit(True);
```

Comment:
Git went... a little crazy with the diff-tracking here.  It looks like these two lines are the main change, and they weren't here before.

There's no comment for these new lines, and appears that meeting the condition will skip a few checks that were previously run - probably the next three if-conditions.  Is this correct and safe?  Just want to double-check.

Reply:
Yeah, this diff is easier to read if you go with ignore-whitespace-changes.

I will add a comment to the code accordingly, but basically, the mitigation check above can return `LangID = 0` for some languages, which means that a transient language needs to be installed. But if `LangID <> 0`, then we already have the installation lang id, so can exit with success.

Reply:
Note also that `BCP47Tag` is now an in/out parameter rather than in only.